MapboxCoreNavigation/RouteController.swift

```diff
@@ -122,30 +138,74 @@ open class RouteController: NSObject {
      */
     public var snapsUserLocationAnnotationToRoute = true
     
-    var lastReRouteLocation: CLLocation?
+    var isRerouting = false
     
     var routeTask: URLSessionDataTask?
     
+    /// :nodoc: This is used internally when the navigation UI is being used
+    public var usesDefaultUserInterface = false
+    
+    var sessionState = SessionState()
+    var outstandingFeedbackEvents = [CoreFeedbackEvent]()
+    
     /**
      Intializes a new `RouteController`.
      
      - parameter route: The route to follow.
      - parameter directions: The Directions object that created `route`.
      - parameter locationManager: The associated location manager.
+     - parameter accessToken: Optional Mapbox accessToken.
      */
-    @objc(initWithRoute:directions:locationManager:)
-    public init(along route: Route, directions: Directions = Directions.shared, locationManager: NavigationLocationManager = NavigationLocationManager()) {
+    @objc(initWithRoute:directions:locationManager:accessToken:)
+    public init(along route: Route, directions: Directions = Directions.shared, locationManager: NavigationLocationManager = NavigationLocationManager(), accessToken: String? = nil) {
         self.directions = directions
         self.routeProgress = RouteProgress(route: route)
         self.locationManager = locationManager
         self.locationManager.activityType = route.routeOptions.activityType
         super.init()
         
         self.locationManager.delegate = self
+        
+        self.resetSession()
```

Comment:
We should move this call to reset the telemetry session just after the telemetry event initialization.